src/secret_connection.rs

```diff
@@ -2,7 +2,7 @@ use byteorder::{BigEndian, ByteOrder};
 use error::Error;
 #[allow(dead_code)]
 use hkdf::Hkdf;
-use hkdfchachapoly::{new_hkdfchachapoly, Aead, TAG_SIZE};
+use ring::aead;
```

Comment:
For what it's worth I'd like to work on a `core::arch`-based ChaCha20 soon, but in the meantime... _ring_ is a good library.